### ISSUE TITLE
regsvr32: Failed to load DLL 'wineasio.dll'

### ISSUE THREAD

**Issue by ciolansteen** [falkTX/wineasio#3]
Distro - ArchLinux 64bit
Wine 5.4
after compiling, I copied the file to /usr/lib/wine/

running regsvr32 wineasio.dll returns regsvr32: Failed to load DLL 'wineasio.dll
Hoever, giving the full path registers successfully but wineasio don't show up in the libraries list.

Tried with both 32 and 64 bit

Am I missing something here?

**Comment by falkTX** [comment]
Yes, fine where wine has its libraries and place it near side all the other *.dll.so files
It is likely not /usr/lib/wine, perhaps /usr/lib32/wine ?

**Comment by ciolansteen** [comment]
OK, that was it regarding the registration. Copied 32bit build /usr/lib32/wine and 64bit to /usr/lib/wine. Now it registers correctly. However, wineasio.dll still does not show up in the libraries list. Tested with GuitarRig 5.22 (which worked well in the past with wine4.x and the old wineasio library). The asio entry shows in audio settings, but cannot be selected.

**Comment by ciolansteen** [comment]
I may be doing something really wrong here. 
So, I tried with - wine-staging 5.4, lutris5.4 and proton5.0
The dll registers successfully. However, wineasio does not show in libraries list.

**Comment by falkTX** [comment]
app installed in custom wine prefix?

**Comment by ciolansteen** [comment]
yes, and wineasio.dll.so installed correctly in each distro's libraries folder (both 32 and 64bit)

**Comment by ChuckDaniels87** [comment]
I have a similar issue in Arch for several months now. Same problem with Guitar Rig 5 and Amplitube. They even crash when I select ASIO driver.

I will try to post some further details in this issue as soon as I gather more details.

**Comment by ciolansteen** [comment]
That sounds like wine/win program compatibility issue. What i was saying is that after wineasio.dll registration, wineasio does not show up in the "New override for library:" field.

**Comment by ciolansteen** [comment]
Did you tried the old wineasio with wine4.x?

**Comment by ChuckDaniels87** [comment]
This setup was working about a year ago. I will try to go back to wine4.x and test it again to check when it broke.

**Comment by ChuckDaniels87** [comment]
I found quite difficult to compile old version of Wine in Arch. I have tried cloning the ABS repo of Wine and going back to Wine 3.21 and 4.0 but they fail to compile in an updated system.

Also, I used PlayOnLinux to install different versions of Wine but I am having problems connecting `wineasio` with these versions. Any suggestion?

About the error, I have run AmpliTube with winedbg. You can check the crash in [this log](https://gist.github.com/ChuckDaniels87/a1ec8da7829033076d573c6c17f43079). It seems it does not connect with Jack, maybe the problem is in that side.

**Comment by distek** [comment]
On Arch as well.

Had success with the following:

`WINEPREFIX=~/your/wine/prefix WINEDLLPATH=/usr/lib64/wine wine64 regsvr32 wineasio.dll`

This was after copying `wineasio.dll.so` to the `WINEDLLPATH` above.

**Comment by ciolansteen** [comment]
again, the dll registers successfully but it doesn't show up in the DLL overrides list and asio is not working even forcefully overriding it.

**Comment by ChuckDaniels87** [comment]
@ciolansteen I am on Arch too, not on Ubuntu. I have just included a related glibc bug for more information.

As far as I know, Wineasio is using Wine headers of the current Wine install in the system, so I supposed that it needed to be recompiled against the correct version to make it work. I have edited the `Makefile.mk`: 
```makefile
PREFIX                = <PATH_TO_MY_HOME>/.PlayOnLinux/wine/linux-amd64/3.21/
SRCDIR                = .
DLLS                  = wineasio.dll
```
Someone can tell if this is the correct method to compile Wineasio using a version of Wine installed via PlayOnLinux/Lutris?

I tried a downgrade using Pacman, I haven't this method used before because usually causes many issues with the dependencies. It seems that the downgrade work fine for `wine` package, thanks for the tip. However, the problem is still present for wine 4.0 too.

Related to the AUR package, I have managed to compile it at the package it correctly, find the [PKGBUILD here](https://gist.github.com/ChuckDaniels87/a84623e21ac6fbeaf56ade01af68fd99).
